packages/blade/src/components/Modal/Modal.web.tsx

```diff
@@ -152,18 +164,25 @@ const Modal = ({
                   modal: true,
                   label: accessibilityLabel,
                 })}
-                maxWidth={makeSize(modalMaxWidth[size])}
+                maxWidth={size === 'full' ? '100%' : makeSize(modalMaxWidth[size])}
                 minWidth={makeSize(modalMinWidth)}
-                maxHeight={modalMaxHeight}
-                width={`calc(100vw - ${makeSize(modalResponsiveScreenGap)})`}
+                maxHeight={modalMaxHeight[size]}
+                width={
+                  size === 'full'
+                    ? `calc(100vw - ${makeSize(modalMargin[size])} - ${makeSize(
+                        modalMargin[size],
+                      )})`
+                    : `calc(100vw - ${makeSize(modalResponsiveScreenGap)})`
+                }
                 borderRadius={modalBorderRadius}
                 backgroundColor="popup.background.subtle"
                 display="flex"
                 flexDirection="column"
-                top="50%"
-                left="50%"
+                top={size === 'full' ? makeSize(modalMargin[size]) : '50%'}
+                left={size === 'full' ? makeSize(modalMargin[size]) : '50%'}
```

Comment:
Why do we have to define different value for full? wouldn't it still be center aligned just full width?